helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/tracker/FilterTracker.kt

```diff
@@ -0,0 +1,118 @@
+@file:Suppress("FunctionName")
+
+package com.algolia.instantsearch.helper.tracker
+
+import com.algolia.instantsearch.helper.searcher.SearcherMultipleIndex
+import com.algolia.instantsearch.helper.searcher.SearcherSingleIndex
+import com.algolia.instantsearch.helper.tracker.internal.FilterDataTracker
+import com.algolia.instantsearch.helper.tracker.internal.InsightsScope
+import com.algolia.instantsearch.helper.tracker.internal.TrackableSearcher
+import com.algolia.instantsearch.insights.Insights
+import com.algolia.search.model.Attribute
+import com.algolia.search.model.filter.Filter
+import com.algolia.search.model.search.Facet
+import kotlinx.coroutines.CoroutineScope
+
+/**
+ * Tracker of filter events insights.
+ */
+public interface FilterTracker {
+
+    /**
+     * Coroutine scope in which all tracking operations are executed.
+     */
+    val trackerScope: CoroutineScope
+
+    /**
+     * Track a filter click event.
+     *
+     * @param filter filter to track
+     * @param customEventName custom event name, overrides the default event name
+     */
+    public fun <F : Filter> trackClick(filter: F, customEventName: String? = null)
```

Comment:
I would say that we change `customEventName` to `eventName`. In iOS, we have `eventName customEventName: String? = nil` which means the label to the developer is `eventName` but the inner local variable used is `customEventName`

Reply:
Also it keeps consistency with `eventName` that is used in other places

Reply:
Totally agree!
Fixed :D